examples/async_tasks/async_compute.rs

```diff
@@ -0,0 +1,121 @@
+/// them later when we're ready to render our Boxes
+fn add_assets(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let box_mesh_handle = meshes.add(Mesh::from(shape::Cube { size: 0.25 }));
+    commands.insert_resource(BoxMeshHandle(box_mesh_handle));
+
+    let box_material_handle = materials.add(Color::rgb(1.0, 0.2, 0.3).into());
+    commands.insert_resource(BoxMaterialHandle(box_material_handle));
+}
+
+/// This system generates tasks simulating computationally intensive
+/// work that potentially spans multiple frames/ticks. A separate
+/// system, handle_tasks, will poll the spawned tasks on subsequent
+/// frames/ticks, and use the results to spawn cubes
+fn spawn_tasks(mut commands: Commands, thread_pool: Res<AsyncComputeTaskPool>) {
+    for x in 0..NUM_CUBES {
+        for y in 0..NUM_CUBES {
+            for z in 0..NUM_CUBES {
+                // Spawn new task on the AsyncComputeTaskPool
+                let task = thread_pool.spawn(async move {
+                    let mut rng = rand::thread_rng();
+                    let start_time = Instant::now();
+                    let duration = Duration::from_secs_f32(rng.gen_range(0.05..0.2));
+                    while Instant::now() - start_time < duration {
+                        // Spinning for 'duration', simulating doing hard
+                        // compute work generating translation coords!
+                    }
+
+                    // Such hard work, all done!
+                    Transform::from_translation(Vec3::new(x as f32, y as f32, z as f32))
+                });
+
+                // Spawn new entity and add our new task as a component
+                commands.spawn().insert(task);
+            }
+        }
+    }
+}
+
+/// This system queries for entities that have our Task<Transform> component. It polls the
+/// tasks to see if they're complete. If the task is complete it takes the result, adds a
+/// new PbrBundle of components to the entity using the result from the task's work, and
+/// removes the task component from the entity.
+fn handle_tasks(
+    mut commands: Commands,
+    mut our_entity_tasks: Query<(Entity, &mut Task<Transform>)>,
+    box_mesh_handle: Res<BoxMeshHandle>,
+    box_material_handle: Res<BoxMaterialHandle>,
+) {
+    our_entity_tasks.for_each_mut(|(entity, mut task)| {
+        if let Some(transform) = future::block_on(future::poll_once(&mut *task)) {
+            // Add our new PbrBundle of components to our tagged entity
+            commands.entity(entity).insert_bundle(PbrBundle {
+                mesh: box_mesh_handle.0.clone(),
+                material: box_material_handle.0.clone(),
+                transform,
+                ..Default::default()
+            });
+
+            // Task is complete, so remove task component from entity
+            commands.entity(entity).remove::<Task<Transform>>();
+        }
+    });
+}
+
+fn main() {
```

Comment:
nit: By convention, we always put `main` at the top of examples.

Reply:
Changed in a5e2753